core/src/main/java/org/apache/calcite/sql/dialect/PrestoSqlDialect.java

```diff
@@ -131,8 +136,30 @@ private static void unparseUsingLimit(SqlWriter writer, @Nullable SqlNode offset
       RelToSqlConverterUtil.specialOperatorByName("APPROX_DISTINCT")
           .unparse(writer, call, 0, 0);
     } else {
-      // Current impl is same with Postgresql.
-      PostgresqlSqlDialect.DEFAULT.unparseCall(writer, call, leftPrec, rightPrec);
+      switch (call.getKind()) {
```

Comment:
Extract them to `createMapValueConstuctor` function is bettter.
> I like single responsibility principle

Reply:
done, extract new function `convertMapValueCall`

Reply:
Why don't we extract more thoroughly?

We can use more parameter like `unparseCall(writer, call, leftPrec, rightPrec)`

Reply:
done, extract map unparse  method `unparseMapValue`